controllers/dbinstance_controller.go

```diff
@@ -0,0 +1,322 @@
+			SkipCAVerify: dbin.Spec.SSLConnection.SkipVerify,
+		}
+	case "percona":
+		if dbin.Spec.Engine != "mysql" {
+			logrus.Errorf("Instance: name=%s - non mysql percona instance not supported", dbin.Name)
+			return errors.New("non mysql percona instance not supported")
+		}
+
+		instance = &dbinstance.Generic{
+			Host:         dbin.Spec.Percona.ServerList[0].Host,
+			Port:         dbin.Spec.Percona.ServerList[0].Port,
+			Engine:       dbin.Spec.Engine,
+			User:         cred.Username,
+			Password:     cred.Password,
+			SSLEnabled:   dbin.Spec.SSLConnection.Enabled,
+			SkipCAVerify: dbin.Spec.SSLConnection.SkipVerify,
+		}
+	default:
+		return errors.New("not supported backend type")
+	}
+
+	info, err := dbinstance.Create(instance)
+	if err != nil {
+		if err == dbinstance.ErrAlreadyExists {
+			info, err = dbinstance.Update(instance)
+			if err != nil {
+				logrus.Errorf("Instance: name=%s failed updating instance - %s", dbin.Name, err)
+				return err
+			}
+		} else {
+			logrus.Errorf("Instance: name=%s failed creating instance - %s", dbin.Name, err)
+			return err
+		}
+	}
+
+	dbin.Status.Info = info
+	return nil
+}
+
+func (r *DbInstanceReconciler) broadcast(ctx context.Context, dbin *kciv1alpha1.DbInstance) error {
+	dbList := &kciv1alpha1.DatabaseList{}
+	err := r.List(ctx, dbList)
+	if err != nil {
+		return err
+	}
+
+	for _, db := range dbList.Items {
+		ref, err := db.GetInstanceRef()
+		if err != nil {
+			continue
+		}
+		if ref.Name == dbin.Name {
+			annotations := db.ObjectMeta.GetAnnotations()
+			if _, found := annotations["checksum/spec"]; found {
+				annotations["checksum/spec"] = ""
+				db.ObjectMeta.SetAnnotations(annotations)
+				err = r.Update(ctx, &db)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *DbInstanceReconciler) createProxy(ctx context.Context, dbin *kciv1alpha1.DbInstance) error {
+	proxyInterface, err := determineProxyTypeForInstance(r.Conf, dbin)
+	if err != nil {
+		if err == ErrNoProxySupport {
+			return nil
+		}
+		return err
+	}
+
+	// create proxy deployment
+	deploy, err := proxy.BuildDeployment(proxyInterface)
+	if err != nil {
+		return err
+	}
+	err = r.Create(ctx, deploy)
+	if err != nil {
+		if k8serrors.IsAlreadyExists(err) {
+			// if resource already exists, update
+			err = r.Update(ctx, deploy)
+			if err != nil {
+				logrus.Errorf("Instance: name=%s failed updating proxy deployment", dbin.Name)
+				return err
+			}
+		} else {
+			// failed to create deployment
+			logrus.Errorf("Instance: name=%s failed creating proxy deployment", dbin.Name)
+			return err
+		}
+	}
+
+	// create proxy service
+	svc, err := proxy.BuildService(proxyInterface)
+	if err != nil {
+		return err
+	}
+	err = r.Create(context.TODO(), svc)
+	if err != nil {
+		if k8serrors.IsAlreadyExists(err) {
+			// if resource already exists, update
+			patch := client.MergeFrom(svc)
+			err = r.Patch(context.TODO(), svc, patch)
```

Comment:
`ctx` same

Reply:
Nice catch again, I tried to do a search for these but somehow I missed a couple in this file